modin/experimental/core/execution/unidist/implementations/pandas_on_unidist/io/sql.py

```diff
@@ -0,0 +1,275 @@
+    lower_bound : int
+        The minimum value to be requested from the `partition_column`.
+    upper_bound : int
+        The maximum value to be requested from the `partition_column`.
+
+    Returns
+    -------
+    bool
+        Whether the given query is distributable or not.
+    """
+    if (
+        (partition_column is not None)
+        and (lower_bound is not None)
+        and (upper_bound is not None)
+    ):
+        if upper_bound > lower_bound:
+            return True
+        else:
+            raise InvalidArguments("upper_bound must be greater than lower_bound.")
+    elif (partition_column is None) and (lower_bound is None) and (upper_bound is None):
+        return False
+    else:
+        raise InvalidArguments(
+            "Invalid combination of partition_column, lower_bound, upper_bound."
+            + "All these arguments should be passed (distributed) or none of them (standard pandas)."
+        )
+
+
+def is_table(engine, sql):
+    """
+    Check if given `sql` parameter is a table name.
+
+    Parameters
+    ----------
+    engine : sqlalchemy.engine.base.Engine
+        SQLAlchemy connection engine.
+    sql : str
+        SQL query to be executed or a table name.
+
+    Returns
+    -------
+    bool
+        Whether `sql` a table name or not.
+    """
+    if inspect(engine).has_table(sql):
+        return True
+    return False
+
+
+def get_table_metadata(engine, table):
+    """
+    Extract all useful data from the given table.
+
+    Parameters
+    ----------
+    engine : sqlalchemy.engine.base.Engine
+        SQLAlchemy connection engine.
+    table : str
+        Table name.
+
+    Returns
+    -------
+    sqlalchemy.sql.schema.Table
+        Extracted metadata.
+    """
+    metadata = MetaData()
+    metadata.reflect(bind=engine, only=[table])
+    table_metadata = Table(table, metadata, autoload=True)
+    return table_metadata
+
+
+def get_table_columns(metadata):
+    """
+    Extract columns names and python types from the `metadata`.
+
+    Parameters
+    ----------
+    metadata : sqlalchemy.sql.schema.Table
+        Table metadata.
+
+    Returns
+    -------
+    OrderedDict
+        Dictionary with columns names and python types.
+    """
+    cols = OrderedDict()
+    for col in metadata.c:
+        name = str(col).rpartition(".")[2]
+        cols[name] = col.type.python_type.__name__
+    return cols
+
+
+def build_query_from_table(name):
+    """
+    Create a query from the given table name.
+
+    Parameters
+    ----------
+    name : str
+        Table name.
+
+    Returns
+    -------
+    str
+        Query string.
+    """
+    return "SELECT * FROM {0}".format(name)
+
+
+def check_query(query):
+    """
+    Check query sanity.
+
+    Parameters
+    ----------
+    query : str
+        Query string.
+    """
+    q = query.lower()
+    if "select " not in q:
+        raise InvalidQuery("SELECT word not found in the query: {0}".format(query))
+    if " from " not in q:
```

Comment:
now this could break on a multi-line query like that:
```
SELECT foo
FROM
bar
```
(see no spaces around `from`? 😉 )

Reply:
let's leave this comment to be fixed in https://github.com/modin-project/modin/issues/5275, in order to simplify code deduplication.